src/python/pants/backend/python/tasks2/setup_py.py

```diff
@@ -594,8 +598,8 @@ def create(target):
     for target in targets:
       create(target)
 
-    executed = {}  # Collected and returned for tests, processed target -> sdist|setup_dir.
     interpreter = self.context.products.get_data(PythonInterpreter)
+    python_dists = self.context.products.get('python_dists')
```

Comment:
We prefer use of `get_data` (and the other `_data` method variant) which allows you to set the product to whatever data structure you like, instead of the older `get`, which forces use of `ProductMapping`, and requires futzing around with target base dirs and other annoyances.

See the docstring on `pants.goal.products.Products` for details.